### PR TITLE
Refactored packages

### LINKED ISSUE #64: Editable package paths
Reopening from [limetext/lime#464](https://github.com/limetext/lime/issues/464)

### PR DESCRIPTION
**Pull request by zoli**
As a whole this pr is about #71 it doesn't resolve all of the issues but it gets us closer to modularity/optionality. Now the core backend doesn't know anything about sublime packages for using them in the frontend we just need to add 
```go
import _ "github.com/limetext/lime/backend/sublime"
```
closes #64 

As there are a lot of commits and changes I'll address the main changes:

1. packages package which most of the code and the package is documented

2. sublime package the code is now much cleaner also we have separate tests

3. removed [lime-packages](https://github.com/limetext/lime-packages) dependency from backend we dont need this submodule for lime-backend anymore

### CODE REVIEW

**Comment by zoli** [comment]
I tested the frontends against this pr they won't crash but they don't load any packages currently because i need to work a bit on [lime-packages](https://github.com/limetext/lime-packages) repo.

**Comment by zoli** [comment]
closes #85

File: lib/editor.go [hunk 1]
```diff
@@ -125,19 +122,29 @@ func GetEditor() *Editor {
 				buffer:  NewBuffer(),
 				scratch: true,
 			},
-			keyInput: make(chan keys.KeyPress, 32),
+			keyInput:    make(chan keys.KeyPress, 32),
+			defaultSet:  new(HasSettings),
+			platformSet: new(HasSettings),
```
Comment (erbridge) [inline lib/editor.go]: These should probably still be `defaultSettings` and `platformSettings` for clarity.

File: lib/editor_test.go [hunk 2]
```diff
@@ -189,6 +183,7 @@ func TestClipboard(t *testing.T) {
 	}
 }
 
+/* TODO: causing panic
```
Comment (erbridge) [inline lib/editor_test.go]: Maybe `Skip` it rather than commenting it out? https://golang.org/pkg/testing/#T.Skip

File: lib/packages/json.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+type JSON struct {
+	filename string
+	marshal  json.Unmarshaler
+}
+
+func NewJSON(filename string, marshal json.Unmarshaler) *JSON {
+	return &JSON{filename: filename, marshal: marshal}
+}
+
+func NewJSONL(filename string, marshal json.Unmarshaler) *JSON {
```
Comment (erbridge) [inline lib/packages/json.go]: I find this name pretty unclear.

File: lib/view_test.go [hunk 4]
```diff
@@ -251,13 +251,13 @@ func TestStress(t *testing.T) {
 	w := ed.NewWindow()
 	defer w.Close()
 
-	v := w.OpenFile("../frontend/termbox/main.go", 0)
+	v := w.OpenFile("./view.go", 0)
```
Comment (erbridge) [inline lib/view_test.go]: This should probably open a test file for reproducibility.

**Comment by zoli** [comment]
All mentioned problems are resolved.

**Pull request merged**